File: Data Structures & Algorithms/is-subsequence/submission-2.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) > len(t):
            return False
        
        if len(s) == 0:
            return True

        sp = 0
        tp = 0
        res = ""
        while tp < len(t):
            if s[sp] != t[tp]:
                tp += 1
            else:
                res += t[tp]
                tp += 1
                if sp + 1 < len(s):
                    sp += 1 
                else:
                    break
        return res == s

File: Data Structures & Algorithms/majority-element/submission-2.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut count = HashMap::new();
        let mut res = 0;
        let mut max_count = 0;

        for num in nums {
            let c = count.entry(num).or_insert(0);
            *c += 1;
            if *c > max_count {
                res = num;
                max_count = *c
            }
        }
        res
    }
}
